bug: cannot find module @stencil/core/internal/client

### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.19.0

### Current Behavior

I consume our component library as a dist-custom-elements in my React project using Vite bundler. Until I used 4.18.3 of @stencil/core, I didn't face this error. Started seeing this from 4.19.0.

### Expected Behavior

This error shouldn't be thrown

### System Info

```shell
System: node 20.12.0
    Platform: darwin (23.5.0)
   CPU Model: Apple M2 (8 cpus)
    Compiler: /Users/xxx/Documents/Projects/yyy/node_modules/@stencil/core/compiler/stencil.js
       Build: 1719433468
     Stencil: 4.19.0 🏄
  TypeScript: 5.4.5
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.31.1
```


### Steps to Reproduce

Ours is a private component library, but install any Stencil-based component library that uses _@stencil/core@4.19.0_, consume a component from _dist-custom-elements_ and try to build the consuming application.

### Code Reproduction URL

NA

### Additional Information

```
FAIL  src/features/Listings/components/DownList/DownList.test.tsx
 ● Test suite failed to run

   Cannot find module '@stencil/core/internal/client' from 'node_modules/comp-lib/dist/components/ou-avatar2.js'

   Require stack:
     node_modules/comp-lib/dist/components/ou-avatar2.js
     node_modules/comp-lib/dist/components/ou-avatar.js
     node_modules/comp-lib-react/dist/src/components/stencil-generated/index.js
     node_modules/comp-lib-react/dist/src/components/index.js
     node_modules/comp-lib-react/dist/src/index.js
     src/features/Listings/components/DownList/DownList.test.tsx
```

sadly this error occurs in an angular + ionic project as well, when testing with jest.

those tests where failing since 4.18.2 (https://github.com/ionic-team/stencil/issues/5786).

With 4.19.1 i get the error of that issue:

`Cannot find module '@stencil/core/internal/client' from '../../node_modules/@ionic/core/components/index.js'`

Part of the require stack:

```SH
Require stack:
      /node_modules/@ionic/core/components/index.js
      /node_modules/@ionic/angular/fesm2022/ionic-angular-common.mjs
      /node_modules/@ionic/angular/fesm2022/ionic-angular-standalone.mjs
```

I changed nothing in the jest.config. So i have something similar to that in there, which was working before.

```JSON
{
  ...
  transformIgnorePatterns: [
    'node_modules/(?!(@ionic/angular|@ionic/core|ionicons|@stencil/core|@angular/*))'
  ],
  ...
}
```

--- a/package.json
+++ b/package.json
@@ -37,10 +37,12 @@
       "types": "./internal/index.d.ts"
     },
     "./internal/client": {
-      "import": "./internal/client/index.js"
+      "import": "./internal/client/index.js",
+      "require": "./internal/client/index.js"
     },
     "./internal/testing": {
-      "import": "./internal/testing/index.js"
+      "import": "./internal/testing/index.js",
+      "require": "./internal/testing/index.js"
     },
     "./internal/testing/*": {
       "import": "./internal/testing/*"
@@ -61,7 +63,8 @@
       "types": "./compiler/stencil.d.ts"
     },
     "./compiler/*": {
-      "import": "./compiler/*"
+      "import": "./compiler/*",
+      "types": "./compiler/*"
     },
     "./sys/node": {
       "import": "./sys/node/index.js",

--- a/package.json
+++ b/package.json
@@ -71,6 +71,15 @@
     "./sys/node/*": {
       "import": "./sys/node/*",
       "require": "./sys/node/*"
+    },
+    "./testing": {
+      "import": "./testing/index.js",
+      "types": "./testing/index.d.ts",
+      "require": "./testing/index.js"
+    },
+    "./testing/*": {
+      "import": "./testing/*",
+      "require": "./testing/*"
     }
   },
   "scripts": {

--- a/test/end-to-end/exportMap/index.js
+++ b/test/end-to-end/exportMap/index.js
@@ -6,6 +6,7 @@ const { h } = require('@stencil/core');
 const { MockDocument } = require('@stencil/core/mock-doc');
 const appData = require('@stencil/core/internal/app-data');
 const { createNodeLogger } = require('@stencil/core/sys/node');
+const { createTesting } = require('@stencil/core/testing');
 
 assert(typeof version === 'string');
 assert(typeof run, 'function');
@@ -13,3 +14,7 @@ assert(typeof h === 'function');
 assert(typeof MockDocument === 'function');
 assert(Object.keys(appData).length === 3);
 assert(typeof createNodeLogger === 'function');
+assert(typeof createTesting === 'function');
+
+console.log(`🎉 All CJS imports successfully resolved!`);
+console.log('✅ passed!\n');

--- a/test/end-to-end/exportMap/index.mts
+++ b/test/end-to-end/exportMap/index.mts
@@ -1,19 +1,24 @@
-import assert from 'node:assert'
+import assert from 'node:assert';
 
-import { run } from '@stencil/core/cli'
-import { version } from '@stencil/core/compiler'
+import { run } from '@stencil/core/cli';
+import { version } from '@stencil/core/compiler';
 import { MockDocument } from '@stencil/core/mock-doc';
 import type { BuildConditionals } from '@stencil/core/internal';
-import { BUILD } from '@stencil/core/internal/app-data'
-import { createNodeLogger } from '@stencil/core/sys/node'
+import { BUILD } from '@stencil/core/internal/app-data';
+import { createNodeLogger } from '@stencil/core/sys/node';
+import { createTesting } from '@stencil/core/testing';
 
-assert(typeof version === 'string')
-version.slice()
-BUILD as BuildConditionals
+assert(typeof version === 'string');
+version.slice();
+BUILD as BuildConditionals;
 
-assert(typeof run, 'function')
-run.call
+assert(typeof run, 'function');
+run.call;
 
-assert(typeof MockDocument === 'function')
-assert(typeof BUILD !== 'undefined')
-assert(typeof createNodeLogger === 'function')
+assert(typeof MockDocument === 'function');
+assert(typeof BUILD !== 'undefined');
+assert(typeof createNodeLogger === 'function');
+assert(typeof createTesting === 'function');
+
+console.log(`🎉 All ESM imports successfully resolved!`);
+console.log('✅ passed!\n');

